var/spack/repos/builtin/packages/py-neo/package.py

```diff
@@ -22,12 +25,20 @@ class PyNeo(PythonPackage):
     version("0.4.1", sha256="a5a4f3aa31654d52789f679717c9fb622ad4f59b56d227dca490357b9de0a1ce")
     version("0.3.3", sha256="6b80eb5bdc9eb4eca829f7464f861c5f1a3a6289559de037930d529bb3dddefb")
 
+    depends_on("python@3.8:", type=("build", "run"), when="@0.3.3:")
```

Comment:
Pretty sure 3.8 wasn't the lower bound until recently

Reply:
Thanks for catching this, introduced in 0.12.0, changed accordingly.